night_mode/stylers.py

```diff
@@ -356,6 +356,27 @@ def _renderPreview(self, browser, cardChanged=False):
         if browser._previewWindow:
             self.app.take_care_of_night_class(web_object=browser._previewWeb)
 
+
+    @wraps(position='around')
+    def buildTree(self, browser, _old):
+        root = _old(browser)
+        if root: # For Anki 2.1.17++
+            return root
+        # ---------------------------
+        # For Anki 2.1.15--
+        root = browser.sidebarTree
+        for item in root.findItems('', Qt.MatchContains | Qt.MatchRecursive):
```

Comment:
I really like this change, but it should be noted that it does come at a somewhat significant performance cost. For a reasonably large tag collection in a popular medical deck, browser load times increased from ≈400ms to ≈800ms for me (Linux, no additional add-ons installed). Since QTreeWidget performance is pretty bad as is on <2.1.17, I'm not sure if adding onto that is good. Maybe a togglable option could be a good compromise? But then again that might be a waste of dev time for an "old" Anki release.

Reply:
Those were part of the original code block that was removed from the previous commit. #93

Reply:
If you still want to improve the performance on 2.1.15, you can try <a href="https://github.com/lovac42/Blitzkrieg/releases">Blitzkrieg II</a>. The code for the sidebar from 17 has been backported to 15 for compatibility and the next release will have proper support for night mode. But 15 will become obsolete eventually, once 17, 18, and 19 (and 20...) are stabalized.

Reply:
I'm fully on the v17, 18, 19 train now (it's been an adventurous ride), but thanks!